free-2.md

```diff
@@ -1,69 +1,168 @@
 # Free project 2
 
 
+
+This project would be something like the functional equivalent of [scratch][scratch.mit.edu], only as a learning tool for textual coding. It would have a split-screen interface with the area for writing code on the left (maybe in a new language, but most likely in a pre-existing one), and on the right would be an image visualiation of the code, with similar color-coded "building blocks" to show exactly what functions are being called where and what's being passed into them. As code is added, there could be a button to "refresh" the visualization area, which would also be updated when the code is executed.
+
+
+
 ## The user and a language
+
 This section describes who the project would serve and why a language might be a
+
 good way to meet their needs.
 
 
+
+This project would be for people who are trying to learn functional programming, either for their first exposure to CS or for someone who's used to imperative programming.
```

Comment:
I too had an idea for a language aimed at learners - it is an interesting niche to design a language for. I feel like in some sense it walks the line between DSL and GPL.